modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java

```diff
@@ -496,25 +496,28 @@ protected void createMaxResultsLayout() {
         maxResultsTextField.setStyleName("c-maxresults-input");
         maxResultsTextField.setMaxLength(4);
         maxResultsTextField.setWidth(theme.get("cuba.gui.Filter.maxResults.width"));
+        maxResultsTextField.addValueChangeListener(this::onMaxResultsChange);
 
         maxResultsLookupField = maxResultsFieldHelper.createMaxResultsLookupField();
         maxResultsLookupField.setStyleName("c-maxresults-select");
+        ((LookupField<Integer>) maxResultsLookupField).addValueChangeListener(this::onMaxResultsChange);
```

Comment:
Lets add `Integer` type to `maxResultsLookupField` and parameterized:
- `MaxResultsFieldHelper#createMaxResultsLookupField()`
- `MaxResultsFieldHelper#setUpMaxResultsLookupField()` methods too. 

Thus there will no need to cast to `LookupField<Integer>`

Reply:
Fixed